app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -84,11 +115,61 @@ class CompanyInterestList extends Component<Props, State> {
         },
       })
       .then(() => {
-        this.props.push(`/companyInterest?semesters=${clickedOption.id}`);
+        this.props.push(
+          `/companyInterest?semesters=${clickedOption.id}&event=${this.state.selectedEvent}`
+        );
       });
   };
+  handleEventChange = (clickedOption: EventOption): void => {
+    const { value } = clickedOption;
+    this.props
+      .fetch({
+        filters: {
+          events: value !== null ? value : null,
+        },
+      })
+      .then(() => {
+        this.props.push(
+          `/companyInterest?semesters=${this.props.selectedSemesterOption.id}&event=${clickedOption.value}`
+        );
+      });
+    this.setState({ selectedEvent: clickedOption.value });
```

Comment:
If you have the event stored in the URL query, do you need it in the state as well? I.e. the query params could act as the state for this just fine

Reply:
Done